src/Microsoft.AspNetCore.Sockets/Internal/ChannelConnection.cs

```diff
@@ -11,20 +11,23 @@ namespace Microsoft.AspNetCore.Sockets.Internal
 {
     public class ChannelConnection<T> : IChannelConnection<T>
     {
-        public IReadableChannel<T> Input { get; }
-        public IWritableChannel<T> Output { get; }
+        public IChannel<T> Input { get; }
```

Comment:
I'm OK with that. I just don't want user code getting access to the full Channel :).